system/security/securesocket/securesocket.cpp

```diff
@@ -164,28 +164,31 @@ class CSecureSocket : implements ISecureSocket, public CInterface
     virtual void   read(void* buf, size32_t size)
     {
         size32_t size_read;
-        readTimeout(buf, size, size, size_read, 0, false);
+        // MCK - this was:
+        // readTimeout(buf, size, size, size_read, 0, false);
+        // but that is essentially a non-blocking read() and we want a blocking read() ...
+        readTimeout(buf, 0, size, size_read, WAIT_FOREVER, false);
```

Comment:
I have verified testsocket and soapplus in secure mode do not always work correctly without this change.
A full QA suite passes, but that doe not really push much securesocket code coverage.  What other securesocket testing should we do ?